firmware-support/bittide-sys/src/mac.rs

```diff
@@ -2,13 +2,20 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+use core::ops::Sub;
 use ufmt::derive::uDebug;
+
+/// Peripheral that keeps track of the status of the MAC.
 pub struct Mac {
     status: *const MacStatus,
 }
 
 impl Mac {
-    pub fn new(addr: usize) -> Mac {
+    /// Creates a new instance of `Mac`.
+    ///
+    /// # Safety
+    /// - `addr` must be the base address of the Mac status peripheral.
+    pub unsafe fn new(addr: usize) -> Mac {
```

Comment:
Please use a `*const MacStatus` instead a `usize` here

```suggestion
    pub unsafe fn new(addr: *const MacStatus) -> Mac {
```